caffe2/core/context_gpu.h

```diff
@@ -104,10 +119,16 @@ class ThreadLocalCUDAObjects {
           CUDA_CHECK(cudaStreamDestroy(stream));
         }
       }
+      for (auto& handle : cudnn_handles_[i]) {
+        if (handle) {
+          CUDNN_CHECK(cudnnDestroy(handle));
```

Comment:
should we do this before destroying stream?

Reply:
Actually this is fine -- `cudnnDestroy` doesn't touch a user-passed stream, so it doesn't matter if it exists or not at this point.